### ISSUE TITLE
Support module specific repository.

### ISSUE THREAD

**Issue by junghoon-vans** [spring-projects/spring-data-redis#2816]
Unlike what the [reference documentation](https://docs.spring.io/spring-data/redis/reference/repositories/definition.html#repositories.multiple-modules.types) says, a module-specific repository is not provided.
so we need to add that functionality to support muliple modules.

**Comment by mp911de** [comment]
This is by design to avoid having another empty store-specific interface. 
Repository disambiguation works also through annotations on the domain type. If you add `@RedisHash` to your entity, you can leverage disambiguation. 

Let me know how this goes.

**Comment by junghoon-vans** [comment]
Of course, as you say, there is a way to use annotations, but according to the [reference documentation](https://docs.spring.io/spring-data/redis/reference/repositories/definition.html#repositories.multiple-modules.types), a module-based repository should also be provided.

However, in the current implementation, there is no repository interface that is specific to Redis, so there are limited ways to utilize a module-based repository.

So I created a mark interface that is simply an extension of KeyValueRepository. This interface is only utilized by Redis, so it is clearly distinguishable from other repositories.

**Comment by mp911de** [comment]
Over time, we've realized that introducing module-specific interface types are a harm to the desired programming model. Again, introducing empty marker interfaces do not improve such arrangements but they introduce technology-specific dependencies whereas Spring Data aims to create a store-independent experience in which calling code shouldn't be aware of the technology-specifics.
